arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1540,22 +1540,20 @@ void ClusterInfo::loadPlan() {
         }
 
         auto shardIDs = newCollection->shardIds();
-        auto shards = std::make_shared<std::vector<ServerID>>();
+        auto shards = allocateShared<std::vector<ShardID>>();
         shards->reserve(shardIDs->size());
         newShardToName.reserve(shardIDs->size());
 
         for (auto const& p : *shardIDs) {
-          TRI_ASSERT(p.first.size() >= 2);
           shards->push_back(p.first);
           auto v = allocateShared<ManagedVector<pmr::ServerID>>();
           v->assign(p.second.begin(), p.second.end());
-          newShardsToPlanServers.insert_or_assign(
-              pmr::ShardID{p.first, _resourceMonitor}, std::move(v));
+          newShardsToPlanServers.insert_or_assign(p.first, std::move(v));
           newShardToName.insert_or_assign(p.first, newCollection->name());
         }
 
         // Sort by the number in the shard ID ("s0000001" for example):
-        ShardingInfo::sortShardNamesNumerically(*shards);
+        std::sort(shards->begin(), shards->end());
```

Comment:
Instead of using a vector and then sorting it, we could use a sorted container here from the get-go.
That may require changing other places too, so please treat this as optional.

Reply:
I agree using a sorted container here would be better.
But this is also used outside of Cluster Info, probably also the reason why _shards is the only one not captured in a Managed vector.

This is some medium size change i would like to avoid right now.